src/pages/CatalogPage/CatalogPage.js

```diff
@@ -1,93 +1,71 @@
+import CategoryMoreButton from '../../components/CategoryMoreButton/CategoryMoreButton';
 import Card from "../../components/Card/Card";
 
-const productsTest = [
-  {
-    id: 1,
-    title: "Parfum Chanel №5",
-    image: 'http://placekitten.com/200/300',
-    alt: "Chanel №5 Perfume",
-    price: 2550,
-  },
-  {
-    id: 2,
-    title: "Product 2",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 2",
-    price: 100,
-  },
-  {
-    id: 3,
-    title: "Product 3",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 3",
-    price: 500,
-  },
-  {
-    id: 4,
-    title: "Product 4",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 4",
-    price: 750,
-  },
-  {
-    id: 5,
-    title: "Product 5",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 5",
-    price: 1200,
-  },
-  {
-    id: 2,
-    title: "Product 2",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 2",
-    price: 100,
-  },
-  {
-    id: 3,
-    title: "Product 3",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 3",
-    price: 500,
-  },
-  {
-    id: 4,
-    title: "Product 4",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 4",
-    price: 750,
-  },
-  {
-    id: 5,
-    title: "Product 5",
-    image: 'http://placekitten.com/200/300',
-    alt: "Product 5",
-    price: 1200,
-  },
-];
+import { products, categories } from '../../data'
+
 
 const CatalogPage = () => {
+  const sm = useMediaQuery('(min-width: 480px)');
+  const md = useMediaQuery('(min-width: 768px)');
+  const desktop = useMediaQuery('(min-width: 1440px)');
+  const [showMore, setShowMore] = useState(false);
+  const [buttonText, setButtonText] = useState('Більше категорій');
+
+  const handleClickCategoryButton = (e) => {
+    return
+  };
+
+  const handleClickShowMore = () => {
+    setShowMore((oldValue) => !oldValue);
+    showMore
+      ? setButtonText('Більше категорій')
+      : setButtonText('Зменшити');
+  };
+
   return (
-    <div className={styles.catalogPageMainContent}>
-      <div className={styles.catalogPageCategory}>
-        {new Array(8).fill(null).map((el, i) => <div key={i} className={styles.testCategory}></div>)}
-        {/* place for Category component */}
+    <Container sx={{ paddingY: 2.5 }}>
+      <Advertising />
+      {desktop ? (
+        <CategoryList
+          array={categories}
+          rowSpacing={{ xl: 6.25 }}
+          columnSpacing={{ xl: 0 }}
+          xlColumns={12}
+          onClick={handleClickCategoryButton} />
+      ) : (
+        <Grid container sx={{ flexDirection: 'column', marginY: 2.5 }}>
+          <CategoryList
+            array={categories.slice(0, showMore ? categories.length : ( md ? 4 : (sm ? 3 : 2)))}
+            rowSpacing={{ xs: 2.5 }}
+            columnSpacing={{ xs: 1 }}
+            xsColumns={6}
+            smColumns={4}
+            mdColumns={3}
+            onClick={handleClickCategoryButton} />
```

Comment:
All these conditions must be inside the CategoryList component. The parent component mustn't be worried about how to show the component on the different screens.